rfcs/0139-declarative-filesystem-layouts.md

```diff
@@ -0,0 +1,66 @@
+---
+feature: declarative-filesystem-layouts
+start-date: 2023-01-11
+author: l0b0
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: https://github.com/NixOS/nixpkgs/issues/209988 (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+One of the hardest parts of setting up a NixOS (or any Linux) system from scratch is setting up the filesystems. Deciding how big each partition needs to be, making sure it's UEFI compliant, whether to use LUKS inside LVM or LVM inside LUKS, remembering to set a partition bootable, etc. It would be fantastic to have a declarative way to deal with this. Features which come to mind include:
+
+- Integrates with the GUI installer. Point the installer to a `configuration.nix` with a disk layout somewhere, and it does all the necessary setup, asking for things like passphrases where necessary.
+- A tool to extract the Nix configuration from currently mounted file systems, expanding on what `nixos-generate-config` already does.
```

Comment:
This actually would be nice, independently of the RFC if it doesn't happen. Would make setting up new hosts a breeze, and would be especially nice if it could somehow grab mounts opts too